tripal/tripal_galaxy
# Major API update to remove SID dependency

This pull request does the following: 

- Adjusts the galaxy workflow tables so they don't require a submission ID through creating a linker table between tripal_galaxy_workflow_submission and webform_submissions. 
- The tripal_galaxy_workflow_submission has been altered so that the sid column becomes a unique serial
- There is an update function that moves the existing SID information into the linker table for backwards compatible
- tripal_galaxy_check_submission_status has been updated to run with or without a passed sid and to work with the new linker table.
- tripal_galaxy_invoke_workflow has been refactored so that it just invokes the workflow, the parameters and additional required arguments are now packaged in tripal_galaxy_invoke_webform_submission in tripal_galaxy_webform.inc. This allows tripal galaxy sites to customize submission behavior and bypass the webform integration.

To test:

1. pull branch tg_sid_update_2
2. run drush updatedb
3. After the update make sure any existing submissions are still in the table on the analyses tab.
4. go through the process of creating a workflow submission and submit it. This should go off without a hitch
5. run drush trp-galaxy-status ensure it updates the status of the submission and successfully completes when no submission id is passed to it.
